File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numDict = new HashMap<>();
        int[] result = {0, 0};

        for (int i = 0; i < nums.length; i++) {
            numDict.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            if (numDict.containsKey(target - nums[i])) {
                if (i != numDict.get(target - nums[i])) {
                    result[0] = i;
                    result[1] = numDict.get(target - nums[i]);
                    break;
                }
            }
        }

        return result;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numDict;
        vector<int> result(2, 0);

        for (int i = 0; i < nums.size(); i++) {
            numDict.insert({nums[i], i});
        }

        for (int i = 0; i < nums.size(); i++) {
            unordered_map<int, int>::const_iterator chk = numDict.find(target - nums[i]);
            if (chk != numDict.end() && chk->second != i) {
                result[0] = i;
                result[1] = chk->second;
                break;
            }
        }

        return result;
    }
};